go.mod

```diff
@@ -0,0 +1,78 @@
+module github.com/kubermatic/machine-controller
+
+go 1.13
+
+require (
+	cloud.google.com/go v0.36.0
+	contrib.go.opencensus.io/exporter/ocagent v0.4.12 // indirect
+	git.apache.org/thrift.git v0.12.0 // indirect
+	github.com/Azure/azure-sdk-for-go v31.1.0+incompatible
+	github.com/Azure/go-autorest v12.3.0+incompatible // indirect
+	github.com/Azure/go-autorest/autorest v0.9.4 // indirect
+	github.com/Azure/go-autorest/autorest/azure/auth v0.4.2
+	github.com/Azure/go-autorest/autorest/to v0.3.0
+	github.com/Azure/go-autorest/autorest/validation v0.2.0 // indirect
+	github.com/Masterminds/semver v1.4.2
+	github.com/Masterminds/sprig v2.15.0+incompatible
+	github.com/ajeddeloh/go-json v0.0.0-20170920214419-6a2fe990e083 // indirect
+	github.com/ajeddeloh/yaml v0.0.0-20170912190910-6b94386aeefd // indirect
+	github.com/aliyun/alibaba-cloud-sdk-go v0.0.0-20190828035149-111b102694f9
+	github.com/aokoli/goutils v1.0.1 // indirect
+	github.com/aws/aws-sdk-go v1.19.25
+	github.com/census-instrumentation/opencensus-proto v0.2.1 // indirect
+	github.com/coreos/container-linux-config-transpiler v0.6.1
+	github.com/coreos/go-semver v0.2.0 // indirect
+	github.com/coreos/ignition v0.24.0 // indirect
+	github.com/davecgh/go-spew v1.1.1
+	github.com/digitalocean/godo v1.1.3
+	github.com/docker/distribution v2.7.1+incompatible
+	github.com/emicklei/go-restful v2.8.0+incompatible // indirect
+	github.com/ghodss/yaml v1.0.0
+	github.com/go-test/deep v1.0.1
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/google/go-cmp v0.3.1 // indirect
+	github.com/gophercloud/gophercloud v0.2.1-0.20190626201551-2949719e8258
+	github.com/grpc-ecosystem/grpc-gateway v1.9.5 // indirect
+	github.com/heptiolabs/healthcheck v0.0.0-20180807145615-6ff867650f40
+	github.com/hetznercloud/hcloud-go v1.15.1
+	github.com/huandu/xstrings v1.0.0 // indirect
+	github.com/linode/linodego v0.7.1
+	github.com/mattbaird/jsonpatch v0.0.0-20171005235357-81af80346b1a
+	github.com/oklog/run v1.0.0
+	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
+	github.com/packethost/packngo v0.1.1-0.20190410075950-a02c426e4888
+	github.com/patrickmn/go-cache v2.1.0+incompatible
+	github.com/pborman/uuid v1.2.0
+	github.com/pkg/errors v0.8.1
+	github.com/pmezard/go-difflib v1.0.0
+	github.com/prometheus/client_golang v1.0.0
+	github.com/sethvargo/go-password v0.1.2
+	github.com/tent/http-link-go v0.0.0-20130702225549-ac974c61c2f9 // indirect
+	github.com/vincent-petithory/dataurl v0.0.0-20160330182126-9a301d65acbb // indirect
+	github.com/vmware/govmomi v0.22.2
+	golang.org/x/crypto v0.0.0-20191206172530-e9b2fee46413
+	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
+	golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898 // indirect
+	gomodules.xyz/jsonpatch v2.0.1+incompatible // indirect
+	google.golang.org/api v0.3.1
+	gopkg.in/gcfg.v1 v1.2.3
+	gopkg.in/ini.v1 v1.46.0 // indirect
+	gopkg.in/warnings.v0 v0.1.2 // indirect
+	k8s.io/api v0.0.0-20191004120003-3a12735a829a
```

Comment:
There is support go mods from upstream: 
https://pkg.go.dev/mod/k8s.io/api@v0.17.3?tab=versions (the earliest version is v0.15.7, for all k8s.io dependencies). This however could potentially conflict with sigs.k8s.io/controller-runtime v0.3.0 go.mod file definition.

Reply:
Ok, I'll give it a try